File: lotto.py
import random

class Lotto:
	def __init__(self, numBalls, numBallsDrawn):
		self.numBalls = numBalls
		self.numBallsDrawn = numBallsDrawn
		self.winner = False
		self.balls = []
		self.week = 0

		for i in range(self.numBalls):
			self.balls.append(i+1)

	def getLottoNumbers(self):
		a = []
		for i in range(self.numBallsDrawn):
			b = random.choice(self.balls)
			while b in a:
				b = random.choice(self.balls)
			a.append(b)
			a.sort()
		return a

	def isWinner(self):
		if self.winner:
			return True
		else:
			return False

	def checkWinner(self):
		theWinners = self.getLottoNumbers()
		quickPick = self.getLottoNumbers()
		
		if theWinners == quickPick:
			self.winner = True
			self.week += 1
			# print('{} {} {} You won!'.format(self.week, theWinners, quickPick))
			return('You won! {} {} {}'.format(self.week, theWinners, quickPick))
		else:
			self.week += 1
			# print('{} {} {}'.format(self.week, theWinners, quickPick))
			return('Not a winner {} {} {}'.format(self.week, theWinners, quickPick))

File: lotto.js
var ballMachine = {
	balls: [],
	numBalls: 3,
	numBallsDrawn: 2,
	populateBalls: function(numBalls) {
		for (var i = 0; i < numBalls; i++) {
			this.balls.push(i+1);
		}
	},
	drawNumbers: function(numBallsDrawn) {
		var a = [];
		var b = [];
		for (var i = 0; i < numBallsDrawn; i++) {
			b = this.getRandomBall(this.balls);
			while (a.includes(b)) {
				b = this.getRandomBall(this.balls);
			}
		a.push(b);
		}
		a.sort();
		return a;
	},
	getRandomBall: function() {
    return Math.floor(Math.random()*(this.balls.length-1+1))+1;
	},
};

var lottoGame = {
	theWinners: [],
	playerNumbers: [],

	init: function() {
		ballMachine.balls = [];
		ballMachine.populateBalls(ballMachine.numBalls);
		theWinners = ballMachine.drawNumbers(ballMachine.numBallsDrawn);
		playerNumbers = ballMachine.drawNumbers(ballMachine.numBallsDrawn);
	},
	isWinner: function() {
		for (var i = 0; i < theWinners.length; i++) {
			if (theWinners[i] != playerNumbers[i]) {
				return false;
			}
		}
		return true;
	},
	displayStuff: function() {
		this.init();
		console.log(ballMachine.balls + '\n' + theWinners + ' -- ' + playerNumbers)
		if (this.isWinner()) {
			console.log('You won!');
		}	else {
			console.log('You lost');
		}
	}
};